src/main/java/at/hannibal2/skyhanni/features/misc/reminders/ReminderManager.kt

```diff
@@ -0,0 +1,256 @@
+        val remindAt = SimpleTimeMark.now().plus(time)
+
+        storage[newId()] = Reminder(reminder, remindAt)
+        sendMessage("§6Reminder set for ${time.format()}")
+    }
+
+    private fun actionReminder(
+        args: List<String>,
+        command: String,
+        vararg arguments: String,
+        action: (List<String>, Reminder) -> String,
+    ) {
+        val argumentText = arguments.joinToString(" ")
+        if (args.size < arguments.size) return ChatUtils.userError("/shremind $command $argumentText")
+
+        if (args.first() == "-l") {
+            if (args.size < arguments.size + 1) return ChatUtils.userError("/shremind $command -l $argumentText")
+            if (storage[args.drop(1).first()] == null) return ChatUtils.userError("Reminder not found")
+            action(args.drop(2), storage[args.drop(1).first()]!!).apply {
+                listReminders(listPage)
+                sendMessage(this)
+            }
+        } else if (storage[args.first()] == null) {
+            return ChatUtils.userError("Reminder not found")
+        } else {
+            sendMessage(action(args.drop(1), storage[args.first()]!!))
+        }
+    }
+
+    private fun removeReminder(args: List<String>) = actionReminder(
+        args,
+        "remove",
+        "[id]",
+    ) { _, reminder ->
+        storage.values.remove(reminder)
+        "§cReminder deleted."
+    }
+
+    private fun editReminder(args: List<String>) = actionReminder(
+        args,
+        "edit",
+        "[id]",
+        "[reminder]",
+    ) { arguments, reminder ->
+        reminder.reason = arguments.joinToString(" ")
+        "§6Reminder edited."
+    }
+
+    private fun moveReminder(args: List<String>) = actionReminder(
+        args,
+        "move",
+        "[id]",
+        "[time]",
+    ) { arguments, reminder ->
+        val time = parseReminder(arguments.first()) ?: return@actionReminder "§cInvalid time format!"
+        reminder.remindAt = SimpleTimeMark.now().plus(time)
+        "§6Reminder moved to ${time.format()}"
+    }
+
+    private fun help() {
+        createDivider().send()
+        "§6SkyHanni Reminder Commands:".asComponent().send()
+        "§e/shremind <time> <reminder> - §bCreates a new reminder".asComponent().send()
+        "§e/shremind list <page> - §bLists all reminders".asComponent().send()
+        "§e/shremind remove <id> - §bRemoves a reminder".asComponent().send()
+        "§e/shremind edit <id> <reminder> - §bEdits a reminder".asComponent().send()
+        "§e/shremind move <id> <time> - §bMoves a reminder".asComponent().send()
+        "§e/shremind help - §bShows this help message".asComponent().send()
+        createDivider().send()
+    }
+
+    @SubscribeEvent
+    fun onSecondPassed(event: SecondPassedEvent) {
+        val remindersToSend = mutableListOf<IChatComponent>()
+
+        for ((id, reminder) in getSortedReminders()) {
+            if (!reminder.shouldRemind(config.interval.minutes)) break
```

Comment:
break feels wrong here. how about continue?

Reply:
its sorted by time so if it cant remind that means all other reminders after this point will also not be able to be reminded of

Reply:
wait actually nvm I forgot about repeat reminders